crates/miden-tx/src/executor/notes_checker.rs

```diff
@@ -51,69 +71,68 @@ where
         input_notes: InputNotes<InputNote>,
         tx_args: TransactionArgs,
         source_manager: Arc<dyn SourceManager>,
-    ) -> Result<NoteAccountExecution, TransactionExecutorError> {
-        // Check input notes
-        // ----------------------------------------------------------------------------------------
-
-        let mut successful_notes = vec![];
-        for note in input_notes.iter() {
+    ) -> Result<NoteConsumptionInfo, TransactionExecutorError> {
+        let input_note_count = input_notes.num_notes() as usize;
```

Comment:
nit: I would rename `input_note_count` to `num_input_notes` - this is the convention we usually use.